maskinporten_api/audit.py

```diff
@@ -0,0 +1,40 @@
+import logging
+from datetime import datetime, timezone
+
+import boto3
+from botocore.exceptions import ClientError
+
+log = logging.getLogger()
+
+
+def audit_log(item_id, item_type, env, action, user, scopes=None, client_id=None):
+    """Add a log entry to the API's audit trail."""
+
+    dynamodb = boto3.resource("dynamodb", region_name="eu-west-1")
+    table = dynamodb.Table("maskinporten-audit-trail")
+
+    try:
+        db_response = table.put_item(
+            Item={
```

Comment:
Trenger egentlig feltene i dynamodb å starte med stor bokstav? Jeg vet at det er tilfellet i noen av de eldre tabellene vi har, men jeg tror at det kan være lurt å ha felt-navn som er snake_case og med feltnavn som gjør at en rad tilsvarer et kode-objekt

Reply:
Tror ikke de må ha det, men er ikke det konvensjonen i DynamoDB? Hva er fordelen eventuelt ved å følge Python-konvensjonen?

Reply:
Serialisering og deserialisering og generelt at ting heter det samme over alt

Reply:
> Tror egentlig ikke DynamoDB har noen bestemt navnekonvensjon på feltene. Det står i hvert fall ikke noe her: https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/HowItWorks.NamingRulesDataTypes.html
> 
> Har heller ikke klart å finne noen andre gode ressurser som anbefaler CamelCase fremfor noe annet. Så tror bare vi har basert oss på noe eksempelkode som tilfeldigvis brukte CamelCase.

Åja, da har jeg bare trodd det basert på eksempler jeg har sett og våre andre tabeller. :sweat_smile: Men foreslår uansett at vi fortsetter stilen vi har kjørt på de andre tabellene for vår egen del, sånn at det blir konsistent (`Id`, `Type` osv).